sdk/cosmos/azure-cosmos/azure/cosmos/_change_feed/change_feed_fetcher.py

```diff
@@ -0,0 +1,199 @@
+# The MIT License (MIT)
+# Copyright (c) 2014 Microsoft Corporation
+
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+"""Internal class for processing change feed implementation in the Azure Cosmos
+database service.
+"""
+import base64
+import copy
+import json
+from abc import ABC, abstractmethod
+from typing import Dict, Any, List
+
+from azure.cosmos import _retry_utility, http_constants, exceptions
+from azure.cosmos._change_feed.change_feed_start_from import ChangeFeedStartFromPointInTime
+from azure.cosmos._change_feed.change_feed_state import ChangeFeedStateV1, ChangeFeedStateV2
+from azure.cosmos.exceptions import CosmosHttpResponseError
+
+# pylint: disable=protected-access
+
+class ChangeFeedFetcher(ABC):
+
+    @abstractmethod
+    def fetch_next_block(self):
+        pass
+
+class ChangeFeedFetcherV1(ChangeFeedFetcher):
+    """Internal class for change feed fetch v1 implementation.
+     This is used when partition key range id is used or when the supplied continuation token is in just simple etag.
+     Please note v1 does not support split or merge.
+
+    """
+    def __init__(
+            self,
+            client,
+            resource_link: str,
+            feed_options: Dict[str, Any],
+            fetch_function):
+
+        self._client = client
+        self._feed_options = feed_options
+
+        self._change_feed_state: ChangeFeedStateV1 = self._feed_options.pop("changeFeedState")
+        if not isinstance(self._change_feed_state, ChangeFeedStateV1):
```

Comment:
Please use a public discriminator attribute rather than instance checks, e.g. something like:
```python
if self._change_feed_state.version == 'v1':
    ...
```
Or coupled with an enum if you prefer:
```python
if self._change_feed_state.version == ChangeFeedVersions.One:
    ...
```

Reply:
updated to use public attribute version + ChangeFeedState enum